Lib/asyncio/tasks.py

```diff
@@ -897,17 +930,41 @@ def callback():
     return future
 
 
-# WeakSet containing all alive tasks.
-_all_tasks = weakref.WeakSet()
+def create_eager_task_factory(custom_task_constructor):
+
+    if "eager_start" not in inspect.signature(custom_task_constructor).parameters:
```

Comment:
I am strong -1 on using any introspection, this is wrong as will reject any factory whose signate cannot be represented by inspect module. This needs to be fixed.

Reply:
I added this, but have no particular love for this code. I actually thought we already removed it as it caused some issues when the factory is a function with variable kwargs.